### PR TITLE
Basic porcelain api

### PR DESCRIPTION
**Pull request by richfitz**
This PR creates a basic porcelain api for outpack

### CODE REVIEW

File: README.Rmd [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  } else {
+    response <- jsonlite::prettify(res$body)
+  }
+
+  template <- "### %s %s\n\n```json%s```"
+  sprintf(template, method, path_label, response)
+}
+
+txt <- vapply(commands, do_response, "")
+cat(paste(txt, collapse = "\n\n"))
```
Comment (r-ash) [inline README.Rmd:59]: This is cool!

**Review by r-ash — APPROVED** [review]
This looks great, nice to see the porcelain comment syntax being used too. 

Looks fine to go to me as a PR, but interested where and how this is going to be used? How will it fit in to orderly.server?

File: R/api.R [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  ## TODO: should we check for use_file_store and require_complete_tree here?
+
+  logger <- porcelain::porcelain_logger(log_level)
+  api <- porcelain::porcelain$new(validate = validate, logger = logger)
+  api$include_package_endpoints(state = list(root = root))
+  api
+}
+
+
+##' @porcelain GET / => json(root)
```
Comment (hillalex) [inline R/api.R:29]: oh wow, so you can generate these routes just with comments? very cool

File: README.Rmd (hunk 1 above)
Comment (hillalex) [reply README.Rmd:59]: agree with Rob, this is so good!

**Review by hillalex — APPROVED** [review]

**Pull request merged**